go/updater/trainingJobUpdater.go

```diff
@@ -48,34 +53,416 @@ type TrainingJobUpdater struct {
+	log.Infof("End to delete TrainingJob namespace=%v name=%v", updater.job.Namespace, updater.job.Name)
+
+	if fault {
+		return fmt.Errorf("delete resource error namespace=%v name=%v", updater.job.Namespace, updater.job.Name)
+	}
+	return nil
+}
+
+func (updater *TrainingJobUpdater) createResource(tp padv1.TrainingResourceType) error {
+	resource := new(v1beta1.ReplicaSet)
+	switch tp {
+	case padv1.MASTER:
+		resource = updater.job.Spec.Master.ReplicaSpec
+	case padv1.PSERVER:
+		resource = updater.job.Spec.Pserver.ReplicaSpec
+	default:
+		return fmt.Errorf("unknow resource")
+	}
+	newResource, err := updater.kubeClient.ExtensionsV1beta1().ReplicaSets(updater.job.Namespace).Get(resource.Name, v1.GetOptions{})
+	if errors.IsNotFound(err) {
+		log.Infof("not found to create namespace=%v name=%v resourceName=%v", updater.job.Namespace, updater.job.Name, resource.Name)
+		newResource, err = updater.kubeClient.ExtensionsV1beta1().ReplicaSets(updater.job.Namespace).Create(resource)
+		if err != nil {
+			updater.status.Phase = padv1.TrainingJobPhaseFailed
+			updater.status.Reason = "Internal error; create resource error:" + err.Error()
+			return err
+		}
+	}
+	if newResource != nil {
+		ticker := time.NewTicker(confirmResourceTicker)
+		defer ticker.Stop()
+		for v := range ticker.C {
+			rs, err := updater.kubeClient.ExtensionsV1beta1().ReplicaSets(updater.job.Namespace).Get(resource.Name, v1.GetOptions{})
+			log.Infof("Current time %v runing pod is %v, resourceName=%v", v.String(), rs.Status.ReadyReplicas, resource.Name)
+			if err != nil && !errors.IsServerTimeout(err) && !errors.IsTooManyRequests(err) {
+				updater.status.Reason = "Internal error; create resource error:" + err.Error()
+				return err
+			}
+			if *resource.Spec.Replicas == 0 {
+				return fmt.Errorf(" trainingjob is deleting, namespace=%v name=%v ", updater.job.Namespace, updater.job.Name)
+
+			}
+			if rs.Status.ReadyReplicas == *resource.Spec.Replicas {
+				log.Infof("Create resource done , namespace=%v name=%v resourceName=%v", updater.job.Namespace, updater.job.Name, resource.Name)
+				return nil
+			}
+		}
+	}
+	return nil
+}
+
+func (updater *TrainingJobUpdater) createTrainer() error {
+	newTrainer, err := updater.kubeClient.BatchV1().Jobs(updater.job.Namespace).Get(updater.job.Spec.Trainer.ReplicaSpec.Name, v1.GetOptions{})
+	if errors.IsNotFound(err) {
+		log.Infof("not found to trainer pserver namespace=%v name=%v", updater.job.Namespace, updater.job.Name)
+		newTrainer, err = updater.kubeClient.BatchV1().Jobs(updater.job.Namespace).Create(updater.job.Spec.Trainer.ReplicaSpec)
+		if err != nil {
+			updater.status.Phase = padv1.TrainingJobPhaseFailed
+			updater.status.Reason = "Internal error; create trainer error:" + err.Error()
+			return err
+		}
+	}
+	if newTrainer != nil {
+		updater.status.Phase = padv1.TrainingJobPhaseRunning
+		updater.status.Reason = ""
+		return nil
+	}
+	return nil
+}
+
+func (updater *TrainingJobUpdater) createTrainingJob() error {
+
```

Comment:
Delete redundant blank line.

Reply:
@Yancey1989  Done and thanks for your comments~